static/src/javascripts/projects/commercial/modules/dfp/lazy-load.js

```diff
@@ -1,69 +1,62 @@
 define([
     'fastdom',
-    'lodash/functions/throttle',
     'common/utils/config',
     'common/utils/detect',
     'common/utils/user-timing',
     'commercial/modules/dfp/dfp-env',
     'commercial/modules/dfp/load-advert',
     'commercial/modules/dfp/performance-logging',
     'commercial/modules/dfp/get-advert-by-id'
-], function (fastdom, throttle, config, detect, userTiming, dfpEnv, loadAdvert, performanceLogging, getAdvertById) {
-    /* nbOfFrames: integer. Number of refresh frames we want to throttle the scroll handler */
-    var nbOfFrames = 6;
-
-    /* durationOfFrame: integer. Number of miliseconds a refresh frame typically lasts */
-    var durationOfFrame = 16;
-
+], function (fastdom, config, detect, userTiming, dfpEnv, loadAdvert, performanceLogging, getAdvertById) {
     /* depthOfScreen: double. Top and bottom margin of the visual viewport to check for the presence of an advert */
     var depthOfScreen = 1.5;
 
-    /* loadQueued: boolean. Set to true when a lazyload task is scheduled */
-    var loadQueued = false;
+    var lazyLoad = dfpEnv.lazyLoadObserve ? onIntersect : onScroll;
 
-    var lazyLoad = throttle(function () {
-        var viewportHeight = detect.getViewport().height;
+    return lazyLoad;
 
-        if( loadQueued ) {
-            return;
-        }
+    function onScroll() {
+        var viewportHeight = detect.getViewport().height;
 
-        loadQueued = true;
-        fastdom.read(function () {
-            loadQueued = false;
-            var lazyLoad = dfpEnv.advertsToLoad
-            .filter(function (advert) {
-                var rect = advert.node.getBoundingClientRect();
-                var isNotHidden = rect.top + rect.left + rect.right + rect.bottom !== 0;
-                var isNotTooFarFromTop = (1 - depthOfScreen) * viewportHeight < rect.bottom;
-                var isNotTooFarFromBottom = rect.top < viewportHeight * depthOfScreen;
-                // load the ad only if it's setting within an acceptable range
-                return isNotHidden && isNotTooFarFromTop && isNotTooFarFromBottom;
-            })
-            .map(function (advert) {
-                return advert.id;
-            });
+        var lazyLoad = dfpEnv.advertsToLoad
+        .filter(function (advert) {
+            var rect = advert.node.getBoundingClientRect();
+            var isNotHidden = rect.top + rect.left + rect.right + rect.bottom !== 0;
+            var isNotTooFarFromTop = (1 - depthOfScreen) * viewportHeight < rect.bottom;
```

Comment:
Coordinates are relative to the top coordinate of the viewport. We want slots 1/2 a screen above or below the viewport to be loaded.
- Below is easy: `offsetBelow = viewportHeight * 1.5` (i.e. 150% from the top of the viewport)
- Above, the same formula, `offsetAbove = viewportHeight * -1.5` would go up "one viewport too far", so `offsetAbove = viewportHeight * (1 - 1.5)` fixes that (i.e. -50% from the top of the viewport)